src/ray/raylet/worker_pool_test.cc

```diff
@@ -1244,78 +1311,127 @@ TEST_F(WorkerPoolTest, StartWorkWithDifferentShimPid) {
   auto actor_id = ActorID::Of(JOB_ID, task_id, 1);
   TaskSpecification java_task_spec = ExampleTaskSpec(
       ActorID::Nil(), Language::JAVA, JOB_ID, actor_id, actor_jvm_options, task_id);
-  ASSERT_EQ(worker_pool_->PopWorker(java_task_spec), nullptr);
+  worker = worker_pool_->PopWorkerSync(java_task_spec);
+  ASSERT_NE(worker, nullptr);
```

Comment:
I believe this change didn't make sense, because this test is aiming to test shim pid mechanism.

Since we integrated startup token mechanism, this case should be removed.